gogs_client/_implementation/http_utils.py

```diff
@@ -11,8 +11,9 @@ class RelativeHttpRequestor(object):
     A thin wrapper around the requests module that allows for endpoint paths
     to be given relative to a fixed base URL
     """
-    def __init__(self, base_url):
+    def __init__(self, base_url, session=requests.Session()):
```

Comment:
actually that's on purpose, because it's more likely you'd want to keep the same shared session object *even between instances*. And BTW, it's matching the original behaviour where you were using the global `requests` context 😉